### PR TITLE
Gui: Fix Touchpad navigation style panning in edit mode

### LINKED ISSUE #16523: [Sketcher] Regression: on entering Edit Mode with TouchPad Nav Style, holding Shift key and the slightest mouse movement makes the object jump in X positive, Y negative
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

On starting FreeCAD, click on the _PartDesignExample_ file to load it, select _Face6_ and in _Sketcher_ click the _New Sketch_ icon and confirm _Plane Face_ and click _OK_. As a Touchpad Nav Style user, my default method for aligning the origin is to hold the left _Shift_ key and move the mouse usually a small amount to align to the centre of the window. 

Having completed a Git Bisect, ~this [commit](https://github.com/FreeCAD/FreeCAD/commit/382ca30d1973658850f53720778bbe4064c8d4bc) which seems quite bizarre~ is the cause of the significant jump of the origin to the most lower right corner of the window with the most minimal mouse movement. The latest AppImage doesn't have this behaviour but of course is packaged very differently to my self compiled build.

It seems to me that there doesn't appear to be any reason why the change couldn't have a `# ifdef FC_OS_MACOSX` condition applied to it, @shaise your input would be appreciated.

### Full version info

```shell
OS: Linux Mint 22 (X-Cinnamon/cinnamon)
Word size of FreeCAD: 64-bit
Version: 1.1.0dev.38739 (Git)
Build type: Release
Branch: main
Hash: f09dfa72d0f40568150d8f5ffb6223a416f0e6ab
Python 3.12.3, Qt 5.15.13, Coin 4.0.2, Vtk 9.1.0, OCC 7.6.3
Locale: English/United Kingdom (en_GB)
Stylesheet/Theme/QtStyle: unset/unset/Qt default
Installed mods: 
  * FreeCAD-themes 2024.9.14
  * Silk 0.1.5
  * sheetmetal 0.4.24
  * Behave-Dark-Colors 0.1.1
  * CfdOF 1.27.5
  * freecad.gears 1.2.0
  * Curves 0.6.45
```

### Subproject(s) affected?

None

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Rexbas**
Fixes #16523.

### CODE REVIEW

**Comment by Syres916** [comment]
Thanks for the efforts, works a treat.

File: src/Gui/NavigationStyle.h [hunk 1]
```diff
@@ -415,6 +415,9 @@ class GuiExport TouchpadNavigationStyle : public UserNavigationStyle {
 
 protected:
     SbBool processSoEvent(const SoEvent * const ev) override;
+
+private:
+    SbBool blockPan {false}; // Used to block the first pan in a mouse movement to prevent big jumps
```
Comment (chennes) [inline src/Gui/NavigationStyle.h:420]: Let's consider making this a static local since it's only used in one method.

Comment (zbynekwinkler) [reply src/Gui/NavigationStyle.h:420]: Making it static would make it global wrt. lifetime - initialized on program start and shared among all instances of the class. I'd not go there. For me it is harder to reason about.

Comment (Rexbas) [reply src/Gui/NavigationStyle.h]: Done

Comment (Rexbas) [reply src/Gui/NavigationStyle.h]: Oh I did not see that second message, I don't really mind but I think @zbynekwinkler has a good point.

Comment (chennes) [reply src/Gui/NavigationStyle.h]: I'm fine with either way, I'm always just a little cautious about maintaining a piece of state like this: as a static local it's guaranteed that no other method is touching it. This method is long enough that I agree with @zbynekwinkler that it's not a sure thing that a static would be an improvement, so if you'd rather not, I'm fine with that.

Comment (Rexbas) [reply src/Gui/NavigationStyle.h:420]: Then I am not making it static.

**Comment by chennes** [comment]
Backported to 1.0

**Pull request merged**